c_binding/lsm_local_disk.c

```diff
@@ -379,6 +483,80 @@ int lsm_local_disk_vpd83_search(const char *vpd83,
     return rc;
 }
 
+int lsm_local_disk_vpd80_get(const char *disk_path, char **vpd80,
+                             lsm_error **lsm_err)
+{
+    char tmp_vpd80[_LSM_MAX_VPD80_ID_LEN];
+    char *trimmed_vpd80;
+    const char *sd_name = NULL;
+    int rc = LSM_ERR_OK;
+    char err_msg[_LSM_ERR_MSG_LEN];
+
+    _lsm_err_msg_clear(err_msg);
+
+    rc = _check_null_ptr(err_msg, 3 /* arg_count */, disk_path, vpd80, lsm_err);
+
+    if (rc != LSM_ERR_OK) {
+        if (vpd80 != NULL)
+            *vpd80 = NULL;
+
+        goto out;
+    }
+
+    *vpd80 = NULL;
+    *lsm_err = NULL;
+
+    if (! _file_exists(disk_path)) {
+        rc = LSM_ERR_NOT_FOUND_DISK;
+        _lsm_err_msg_set(err_msg, "Disk %s not found", disk_path);
+        goto out;
+    }
+
+    if (strncmp(disk_path, "/dev/sd", strlen("/dev/sd")) != 0) {
+        rc = LSM_ERR_NO_SUPPORT;
+        _lsm_err_msg_set(err_msg, "we only support disk path start with "
+                         "'/dev/sd' today");
+        goto out;
+    }
+
+    sd_name = disk_path + strlen("/dev/");
+
+    rc = _sysfs_vpd80_serial_num_of_sd_name(err_msg, sd_name, tmp_vpd80);
+    if (rc != LSM_ERR_OK)
+        goto out;
+
+    if (tmp_vpd80[0] != '\0') {
+        trimmed_vpd80 = _trim_spaces(tmp_vpd80);
```

Comment:
Since you don't pass string length to `_trim_spaces()`, please add `tmp_vpd80[_LSM_MAX_VPD80_ID_LEN - 1] = '\0';` before invoke `_trim_spaces()` to make sure passing string is `\0` terminated.